src/Components/Components/src/Forms/FieldIdentifier.cs

```diff
@@ -0,0 +1,111 @@
+    /// model object, or can be any other named value.
+    /// </summary>
+    public readonly struct FieldIdentifier
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FieldIdentifier"/> structure.
+        /// </summary>
+        /// <param name="accessor">An expression that identifies an object member.</param>
+        public static FieldIdentifier Create<T>(Expression<Func<T>> accessor)
+        {
+            ParseAccessor(accessor, out var model, out var fieldName);
+            return new FieldIdentifier(model, fieldName);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FieldIdentifier"/> structure.
+        /// </summary>
+        /// <param name="model">The object that owns the field.</param>
+        /// <param name="fieldName">The name of the editable field.</param>
+        public FieldIdentifier(object model, string fieldName)
+        {
+            if (model == null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+
+            if (model.GetType().IsValueType)
+            {
+                throw new ArgumentException("The model must be a reference-typed object.", nameof(model));
+            }
+
+            Model = model;
+
+            // Note that we do allow an empty string. This is used by some validation systems
+            // as a place to store object-level (not per-property) messages.
+            FieldName = fieldName ?? throw new ArgumentNullException(nameof(fieldName));
+        }
+
+        /// <summary>
+        /// Gets the object that owns the editable field.
+        /// </summary>
+        public object Model { get; }
+
+        /// <summary>
+        /// Gets the name of the editable field.
+        /// </summary>
+        public string FieldName { get; }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+            => (Model, FieldName).GetHashCode();
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+            => obj is FieldIdentifier otherIdentifier
+            && otherIdentifier.Model == Model
+            && string.Equals(otherIdentifier.FieldName, FieldName, StringComparison.Ordinal);
+
+        private static void ParseAccessor<T>(Expression<Func<T>> accessor, out object model, out string fieldName)
+        {
+            var accessorBody = accessor.Body;
+
+            // Unwrap casts to object
+            if (accessorBody is UnaryExpression unaryExpression
+                && unaryExpression.NodeType == ExpressionType.Convert
+                && unaryExpression.Type == typeof(object))
+            {
+                accessorBody = unaryExpression.Operand;
+            }
+
+            if (!(accessorBody is MemberExpression memberExpression))
+            {
+                throw new ArgumentException("The accessor is not supported because its body is not a MemberExpression");
```

Comment:
I'd expect that this is going to see some more clarity. It makes sense to me but I speak compiler.

Reply:
The message is only designed so that people can paste it in GitHub issues and we can easily see which parts of the logic would have to be updated to support a given expression type. I don't expect people to follow it beyond that.

Reply:
I can appreciate that but I think we can do better. It's really common for people to put in some random object where this construct is used in MVC (`() => new Person()').

What if said something like the following:
```
The provided expression contains a {accessorBody.GetType().Name} which is not supported. FieldIdentifier only support simple member accessors (fields, properties) of the object passed into the EditForm.
```

Reply:
OK, that sounds good. Done. Also I found that the expression parser could easily be generalised significantly (or rather, it was a bug that it wasn't already as general as expected), so did that plus tests to demonstrate broader cases.

Reply:
> of the object passed into the EditForm

`of an object` (can be any object; doesn't have to be editForm.Model).